### PR TITLE
Skip when index_columns is empty. Add failing test, bump V and changelog

### LINKED ISSUE #25: When index_columns is not specified, the create table statement has a syntax error.
"Java::JavaSql::SQLSyntaxErrorException: Syntax error: Encountered \"NULL\" at line 1, column 45."
However `index_columns` is optional.
```
      {
        name => "servers"
        columns => [
          ["ip", "varchar(15)"],
          ["descr", "varchar(255)"]
        ]
      }
```
should work but it fails.

In db_object.rb, line 19 needs to change to.
`schema_gen.index(@index_columns) unless @index_columns.empty?`

### PR DESCRIPTION
**Pull request by guyboertje**
Fixes #25

### CODE REVIEW

**Review by jakelandis — APPROVED** [review]
Changes LGTM.

However, may want to consider adding a warning to the log. For any non trivial data set, I would expect the lack of an index to cause a performance issue.

**Comment by guyboertje** [comment]
added log message

File: lib/logstash/filters/jdbc/read_write_database.rb [hunk 1]
```diff
@@ -31,6 +31,9 @@ def build_db_object(db_object)
       begin
         @rwlock.writeLock().lock()
         db_object.build(@db)
+        if db_object.index_columns.empty?
+          logger.warn("db_object '#{db_object.name}': `index_columns` is optional but on larger datasets consider adding an index on the lookup column, it will improve performance")
```
Comment (jakelandis) [inline lib/logstash/filters/jdbc/read_write_database.rb]: nit: remove `db_object` from message. 

nit: a warning should be a bit less friendly :)  suggested:  '#{db_object.name}': `index_columns` has not been defined and may cause performance issues on larger datasets.

Comment (guyboertje) [reply lib/logstash/filters/jdbc/read_write_database.rb]: Agree with less friendly message.

Well, looking at the final line logged, without `do_object` or perhaps `local_db_object`, is it a bit sparse and unhelpful.
```
[2018-05-14T17:04:18,842][WARN ][logstash.filters.jdbc.readwritedatabase] 'servers': `index_columns` has not been defined and may cause performance issues on larger datasets.
```
vs
```
[2018-05-14T17:04:18,842][WARN ][logstash.filters.jdbc.readwritedatabase] local_db_object 'servers': `index_columns` has not been defined and may cause performance issues on larger datasets.
```

Comment (jakelandis) [reply lib/logstash/filters/jdbc/read_write_database.rb]: +1 local_db_object since it matches the config and doc and provides that context.

**Pull request merged**